ui/src/app/shared/services/applications-service.ts

```diff
@@ -20,32 +21,50 @@ function optionsToSearch(options?: QueryOptions) {
     return {};
 }
 
+function getQuery(projects: string[], isListOfApplications: boolean, options?: QueryOptions): any {
+    if (isListOfApplications) {
+        return {projects, ...optionsToSearch(options)};
+    } else {
+        return {...optionsToSearch(options)};
+    }
+}
+
 export class ApplicationsService {
     constructor() {}
 
-    public list(projects: string[], options?: QueryOptions): Promise<models.ApplicationList> {
+    public list(projects: string[], objectListKind: string, options?: QueryOptions): Promise<models.AbstractApplicationList> {
+        const isListOfApplications = objectListKind === 'application';
```

Comment:
I think we should use the objectListKind passed from the caller, instead of hardcoding it, so that we stay as close to the end state as possible.